### PR TITLE
0.2.x Add support for Kotlin flows

### PR DESCRIPTION
**Pull request by a-manumohan**
Adds a new adapter **FlowStreamAdapter** to _scarlet-stream-adapter-coroutines_ module which can be used instead of **ReceiveChannelStreamAdapter** to support operations.

### CODE REVIEW

**Comment by a-manumohan** [comment]
Please accept the license agreement for api 29 as mentioned below 
```
* What went wrong:
A problem occurred configuring project ':scarlet-lifecycle-android'.
> Failed to install the following Android SDK packages as some licences have not been accepted.
     platforms;android-29 Android SDK Platform 29
  To build this project, accept the SDK license agreements and install the missing components using the Android Studio SDK Manager.
  Alternatively, to transfer the license agreements from one workstation to another, see http://d.android.com/r/studio-ui/export-licenses.html
  
  Using Android SDK: /opt/android/sdk
```

File: build.gradle [hunk 1]
```diff
@@ -10,7 +10,7 @@ buildscript {
     }
     dependencies {
         classpath 'com.vanniktech:gradle-maven-publish-plugin:0.8.0'
-        classpath 'com.android.tools.build:gradle:3.4.2'
+        classpath 'com.android.tools.build:gradle:3.5.1'
```
Comment (gs-ts) [inline build.gradle]: I think the update of gradle version and the targetSdkVersion/compileSdkVersion could troubles in such a big library

Comment (etonotieno) [reply build.gradle]: I don't think that should cause an issue. I checked the circleci config file and it hasn't been updated to match the SDK version in the build.gradle file. If changing the circle ci image to 
`docker:`
    `- image: circleci/android:api-29` 
doesn't help, you can add a step to accept the permissions. Example:
`
command: |`
      `yes | sdkmanager --licenses || true
      sdkmanager "platform-tools" "platforms;android-29"
`

Comment (a-manumohan) [reply build.gradle]: Thanks for the review. docker image indeed was the problem. Upgraded to api-29

**Comment by aaronhe42** [comment]
do you mind separating tools version update and adding flow adapter into different PRs?

**Comment by a-manumohan** [comment]
@aaronhe42 makes sense. I have reverted all version upgrades except that of Kotlin coroutines. I will create another PR once this is merged.

**Pull request merged**